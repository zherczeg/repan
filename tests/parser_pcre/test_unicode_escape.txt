# Test unicode escape sequences

# Unicode property support

/\pL\p{L}\P{Cc}\p{L&}\p{SignWriting}/u
    print

/\pA/u
    print

/\p{Unknown}/u
    print

/\p#/u
    print

/\p{Bam!um}/u
    print

/\P{Bamum/u
    print

/\P{sc=Ol_Chiki}\p{Script=o_lchIK_I}/u
    print

/\p{  -S-  _C_  =  -c-  _o_  -P-  _t_  -i-  _C_  }/u
    print

/\P{ Script Extensions = caNs }\p{SCx= AR_mi}/u
    print

/\p{ G-c =  l & }\P{ geneRAl_cate_gor-y = C }/u
    print

/\p{ Script = So }/u
    print

/\p{ Script_Extensions = Pi }/u
    print

/\p{ General_Category = BOPO }/u
    print

/\p{ gc = gc = S }/u
    print

/\p{ gc = a! }/u
    print

/\p{Control}\P{c-o-nt_r_oL}\p{c_nTr-l}/u
    print

/\p{lC}\P{_L_c_}\p{Cased_Letter}\P{ca_sEDLetTe-r}/u
    print

/\p{punctU ation}\P{pu-n-ct}\p{Final_Punctuation}\P{daShpunc Tuation}/u
    print

# Unicode name support

/\N{MAR   Chen-LET --- teRA}\N{ma  -RCHENLETTER-A}\N{marchenletTer   -a}/u
    ascii, print

/\N{hanGU LJUNGSE  ONG--OE}\N{H-A N G-U L J U N-G S E O N G O-  E}/u
    ascii, print

/\N{LAT-I  N SmALl -  - LETTER--N}\N{VARIATION SELECTOR 256}/u
    ascii, print

/\N{A- B -A  Cu-S}\N{Z-o M--b  I  - -e}/u
    ascii, print

/\N{FormFeed}\N{-f-f-}\N{LatIncapITAL-LEt tERGH A}/u
    ascii, print

/\N{ZOMBIEE}/u
    print

/\N{a+b}/u
    print

/\N{zombie/u
    print
