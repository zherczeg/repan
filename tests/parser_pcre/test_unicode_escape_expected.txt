# Test unicode escape sequences

# Unicode property support

/\pL\p{L}\P{Cc}\p{L&}\p{SignWriting}/u
    print
    /\p{L}\p{L}\P{Cc}\p{L&}\p{SignWriting}/

/\pA/u
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\pA/

/\p{Unknown}/u
    print
    Pattern parse failed: Unknown \p or \P property
        /[ HERE ]>>\p{Unknown}/

/\p#/u
    print
    Pattern parse failed: A valid unicode property required after \p or \P
        /[ HERE ]>>\p#/

/\p{Bam!um}/u
    print
    Pattern parse failed: A valid unicode property required after \p or \P
        /[ HERE ]>>\p{Bam!um}/

/\P{Bamum/u
    print
    Pattern parse failed: A right brace (}) is expected
        /\P{Bamum[ HERE ]>>/

/\P{sc=Ol_Chiki}\p{Script=o_lchIK_I}/u
    print
    /\P{Ol_Chiki}\p{Ol_Chiki}/

/\p{  -S-  _C_  =  -c-  _o_  -P-  _t_  -i-  _C_  }/u
    print
    /\p{Coptic}/

/\P{ Script Extensions = caNs }\p{SCx= AR_mi}/u
    print
    /\P{Script_Extensions=Canadian_Aboriginal}\p{Script_Extensions=Imperial_Aramaic}/

/\p{ G-c =  l & }\P{ geneRAl_cate_gor-y = C }/u
    print
    /\p{L&}\P{C}/

/\p{ Script = So }/u
    print
    Pattern parse failed: Unicode script name expected
        /\p{ Script =[ HERE ]>> So }/

/\p{ Script_Extensions = Pi }/u
    print
    Pattern parse failed: Unicode script name expected
        /\p{ Script_Extensions =[ HERE ]>> Pi }/

/\p{ General_Category = BOPO }/u
    print
    Pattern parse failed: Unicode cathegory expected
        /\p{ General_Category =[ HERE ]>> BOPO }/

/\p{ gc = gc = S }/u
    print
    Pattern parse failed: A valid unicode property required after \p or \P
        /[ HERE ]>>\p{ gc = gc = S }/

/\p{ gc = a! }/u
    print
    Pattern parse failed: A valid unicode property required after \p or \P
        /[ HERE ]>>\p{ gc = a! }/

/\p{Control}\P{c-o-nt_r_oL}\p{c_nTr-l}/u
    print
    /\p{Cc}\P{Cc}\p{Cc}/

/\p{lC}\P{_L_c_}\p{Cased_Letter}\P{ca_sEDLetTe-r}/u
    print
    /\p{L&}\P{L&}\p{L&}\P{L&}/

/\p{punctU ation}\P{pu-n-ct}\p{Final_Punctuation}\P{daShpunc Tuation}/u
    print
    /\p{P}\P{P}\p{Pf}\P{Pd}/

# Unicode name support

/\N{MAR   Chen-LET --- teRA}\N{ma  -RCHENLETTER-A}\N{marchenletTer   -a}/u
    ascii, print
    /\x{11c8f}\x{11c88}\x{11c88}/

/\N{hanGU LJUNGSE  ONG--OE}\N{H-A N G-U L J U N-G S E O N G O-  E}/u
    ascii, print
    /\x{116c}\x{1180}/

/\N{LAT-I  N SmALl -  - LETTER--N}\N{VARIATION SELECTOR 256}/u
    ascii, print
    /n\x{e01ef}/

/\N{A- B -A  Cu-S}\N{Z-o M--b  I  - -e}/u
    ascii, print
    /\x{1f9ee}\x{1f9df}/

/\N{FormFeed}\N{-f-f-}\N{LatIncapITAL-LEt tERGH A}/u
    ascii, print
    /\f\f\x{1a2}/

/\N{ZOMBIEE}/u
    print
    Pattern parse failed: Unknown unicode codepoint name
        /\N{[ HERE ]>>ZOMBIEE}/

/\N{a+b}/u
    print
    Pattern parse failed: A right brace (}) is expected
        /\N{a[ HERE ]>>+b}/

/\N{zombie/u
    print
    Pattern parse failed: A right brace (}) is expected
        /\N{zombie[ HERE ]>>/
